exp/addons/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -0,0 +1,115 @@
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// +kubebuilder:rbac:groups=addons.cluster.x-k8s.io,resources=*,verbs=get;list;watch;create;update;patch;delete
+
+// ClusterResourceSetBindingReconciler reconciles a ClusterResourceSetBinding object
+type ClusterResourceSetBindingReconciler struct {
+	Client client.Client
+	Log    logr.Logger
+}
+
+func (r *ClusterResourceSetBindingReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	_, err := ctrl.NewControllerManagedBy(mgr).
+		For(&addonsv1.ClusterResourceSetBinding{}).
+		Watches(
+			&source.Kind{Type: &clusterv1.Cluster{}},
+			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.clusterToClusterResourceSetBinding)},
+		).
+		WithOptions(options).
+		WithEventFilter(predicates.ResourceNotPaused(r.Log)).
+		Build(r)
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
+	return nil
+}
+
+func (r *ClusterResourceSetBindingReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
+	ctx := context.Background()
+
+	// Fetch the ClusterResourceSetBinding instance.
+	binding := &addonsv1.ClusterResourceSetBinding{}
+	if err := r.Client.Get(ctx, req.NamespacedName, binding); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	cluster, err := util.GetOwnerCluster(ctx, r.Client, binding.ObjectMeta)
+	// If the owner cluster is already deleted or in deletion process, delete its ClusterResourceSetBinding
+	if apierrors.IsNotFound(err) || !cluster.DeletionTimestamp.IsZero() {
+		err := r.Client.Delete(ctx, binding)
+		return ctrl.Result{}, err
+	}
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// clusterToClusterResourceSetBinding is mapper function that maps clusters to ClusterResourceSetBinding
+func (r *ClusterResourceSetBindingReconciler) clusterToClusterResourceSetBinding(o handler.MapObject) []ctrl.Request {
+	cluster, ok := o.Object.(*clusterv1.Cluster)
+	if !ok {
+		r.Log.Error(nil, fmt.Sprintf("Expected a Cluster but got a %T", o.Object))
+		return nil
+	}
+
+	crsBinding := &addonsv1.ClusterResourceSetBinding{}
+	if err := r.Client.Get(context.Background(), types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, crsBinding); err != nil {
+		if !apierrors.IsNotFound(err) {
+			r.Log.Error(err, "failed to get ClusterResourceSetBinding")
+		}
+		return nil
+	}
+	return []reconcile.Request{
+		{
+			NamespacedName: client.ObjectKey{
+				Namespace: crsBinding.Namespace,
+				Name:      crsBinding.Name,
+			},
+		},
+	}
```

Comment:
Since the ClusterResourceSetBinding name/namespace matches and we already do a check in `reconcile` for if the resource exists, ignoring `IsNotFound` errors, I think it's probably safe to skip this get all together and just return the request based on the passed in MapObject:
```suggestion
	return []reconcile.Request{
		{
			NamespacedName: client.ObjectKey{
				Namespace: o.Meta.GetNamespace(),
				Name:      o.Meta.GetName(),
			},
		},
	}
```